packages/angular/build/src/tools/angular/compilation/aot-compilation.ts

```diff
@@ -385,3 +422,16 @@ function findAffectedFiles(
 
   return affectedFiles;
 }
+
+function hasOnlyTemplates(modifiedFiles: Set<string>): boolean {
+  for (const file of modifiedFiles) {
+    const lowerFile = file.toLowerCase();
+    if (lowerFile.endsWith('.html') || lowerFile.endsWith('.svg')) {
```

Comment:
NIT: 

```suggestion
    const ext = extname(file).toLowerCase();
    if (ext === '.html' || ext === '.svg') {
```

Or also

```js
 return [...modifiedFiles].every(file => /\.(html|svg)$/i.test(file));
```